### PR TITLE
Create subnet cli

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Create subnet actor command line handler.

This is just a direct json rpc call to ipc agent json rpc server.

Update: Add global config params so that config path will be shared by all cli commands.

### CODE REVIEW

File: src/cli/commands/create.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        let params = CreateSubnetParams {
+            parent: arguments.parent.clone(),
+            name: arguments.name.clone(),
+            min_validator_stake: arguments.min_validator_stake,
+            min_validators: arguments.min_validators,
+            finality_threshold: arguments.finality_threshold,
+            check_period: arguments.check_period,
+        };
+
+        let url = arguments.ipc_agent_url.parse()?;
```
Comment (adlrocha) [inline src/cli/commands/create.rs:34]: Can we fallback to the default one if this isn't set?

File: src/cli/commands/create.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        let json_rpc_client = JsonRpcClientImpl::new(url, None);
```
Comment (adlrocha) [inline src/cli/commands/create.rs:35]: We may need this for all other functions and we may be able to take it out to a common func.

File: src/config/server.rs [hunk 3]
```diff
@@ -9,3 +9,7 @@ pub const JSON_RPC_ENDPOINT: &str = "json_rpc";
 pub struct Server {
     pub json_rpc_address: SocketAddr,
 }
+
+pub mod json_rpc_methods {
+    pub const CREATE_SUBNET: &str = "create_subnet";
```
Comment (adlrocha) [inline src/config/server.rs:14]: Opened a poll in Slack to agree on what convention we want to follow. Let's wait for the results to change/keep this.

**Review by adlrocha — APPROVED** [review]
LGTM. Wait for the RPC naming convention to change it before merging. Thanks 🙏

File: src/cli/commands/create.rs (hunk 1 above)
Comment (cryptoAtwill) [reply src/cli/commands/create.rs:34]: do we have a default value? There is a config file, but the caller can specify any `ipc_agent_url`, should we just let the user config?

File: src/cli/commands/create.rs (hunk 2 above)
Comment (cryptoAtwill) [reply src/cli/commands/create.rs:35]: If I understand correctly, in this CLI, it's only a json rpc client initialization. The `JsonRpcClientImpl::new` is already the most simple form. Or are you referring to the initialization of LotusJsonRpcClient?

**Comment by cryptoAtwill** [comment]
@adlrocha The naming conventions for json rpc methods are updated. Please take a look at other comments.

File: src/cli/commands/create.rs (hunk 1 above)
Comment (adlrocha) [reply src/cli/commands/create.rs:34]: We are setting a default path for config and optionally we can pass a config path also as argument.

**Review by adlrocha — APPROVED** [review]

File: src/cli/commands/create.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+fn get_ipc_agent_url(
+    ipc_agent_url: &Option<String>,
+    global: &GlobalParams,
+    subnet_str: &str,
+) -> Result<Url> {
+    let url = match ipc_agent_url {
+        Some(url) => url.parse()?,
+        None => {
```
Comment (adlrocha) [inline src/cli/commands/create.rs:79]: I don't get what you are doing here, why are we getting the subnet url here? We should be getting the one from the agent not one for the subnet.

File: src/cli/mod.rs [hunk 5]
```diff
@@ -22,5 +26,30 @@ pub trait CommandLineHandler {
...
+    )]
+    config_path: Option<String>,
+}
+
+impl GlobalParams {
+    pub fn config_path(&self) -> String {
+        self.config_path.clone().unwrap_or_else(|| {
+            let home = match std::env::var("HOME") {
+                Ok(home) => home,
+                Err(_) => String::from(""), // cannot get home, just use current dir
```
Comment (adlrocha) [inline src/cli/mod.rs:45]: I don't think this would lead to an accurate value, right?

**Review by adlrocha — CHANGES_REQUESTED** [review]
The new change for the config path LGTM, but have a look to my comments as I don't know if we introduced slight bugs.

File: src/cli/mod.rs (hunk 5 above)
Comment (cryptoAtwill) [reply src/cli/mod.rs:45]: as discussed, should panic!

**Review by adlrocha — APPROVED** [review]
Thank you for the changes, feel free to merge.

**Pull request merged**